### PR TITLE
CAS-641 Extend bedspace search to include number of booked bedspaces

### PR DESCRIPTION
**Pull request by muhammad-elabdulla**
This [PR CAS-641](https://dsdmoj.atlassian.net/browse/CAS-641) is to extends the bedspace search Api endpoint response to include the number of booked bedspaces

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/BedSearchTest.kt [hunk 1]
```diff
@@ -228,7 +228,7 @@ class BedSearchTest : IntegrationTestBase() {
                 resultsPremisesCount = 1,
                 resultsBedCount = 1,
                 results = listOf(
-                  createTemporaryAccommodationBedSearchResult(premises, room, bed, searchPdu.name, 1, listOf(), listOf()),
+                  createTemporaryAccommodationBedSearchResult(premises, room, bed, searchPdu.name, 1, 0, listOf(), listOf()),
```
Comment (davidatkinsuk) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/BedSearchTest.kt:231]: i appreciate it's probably a pain to fix but there's a lot of nondescript arguments here (and in other tests) that i think would be helped by specifying argument names for clarity? something to consider going forward maybe

**Review by davidatkinsuk — APPROVED** [review]
LGTM with a few minor comments 👍

File: src/main/resources/static/_shared.yml [hunk 2]
```diff
@@ -4337,6 +4337,9 @@ components:
         bedCount:
           type: integer
           description: the total number of Beds in the Premises
+        bookedBedCount:
+          type: integer
+          description: the total number of Beds in the Premises
```
[the remark this answers is not in the document]
Comment (muhammad-elabdulla) [reply src/main/resources/static/_shared.yml:4342]: I am sure that I added the right description I don't know how the description end like that  :(

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/BedSearchTest.kt (hunk 1 above)
Comment (muhammad-elabdulla) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/BedSearchTest.kt:231]: I agree I will change it by adding the names of arguments

**Pull request merged**